### PR TITLE
Fix attributte table - order by field type number Fix #1156

### LINKED ISSUE #1156: attribute table - order by field type number
hi

when i try to sort the table by gid, lizmap not parsing field number as number but as string, from postgreSQL tables

imagem

it's a regression...

lwc: all versions
qgis server 2.18 and 3.4

thanks

### PR DESCRIPTION
**Pull request by nboisteault**
With QGIS Server 3.4 DescribeFeatureType requests return 'int' and not 'integer' for integer type.

### CODE REVIEW

**Comment by rldhont** [comment]
It is the same in last QGIS 2.18 version

**Comment by rldhont** [comment]
The Qgis server code to describe type is:
* https://github.com/qgis/QGIS/blob/release-2_18/src/server/qgswfsprojectparser.cpp#L451 for last QGIS Server 2.18.*
* https://github.com/qgis/QGIS/blob/release-3_4/src/server/services/wfs/qgswfsdescribefeaturetype.cpp#L281 for QGIS Server 3.4.* and upper

Numbers can be :
* int
* unsignedInt
* long
* unsignedLong
* integer
* decimal

File: lizmap/www/js/attributeTable.js [hunk 1]
```diff
@@ -1297,7 +1297,7 @@ var lizAttributeTable = function() {
                     // Check if we need to replace url or media by link
                     // Add function for any string cell
                     // First check if the col is number
-                    if (idx in cTypes && cTypes[idx] == 'integer')
+                    if (idx in cTypes && (cTypes[idx] == 'integer' || cTypes[idx] == 'int'))
```
Comment (rldhont) [inline lizmap/www/js/attributeTable.js]: Can you rebuild this by using switch. It will be easiest to cover all the numbers description.

Comment (nboisteault) [reply lizmap/www/js/attributeTable.js]: Do we want 'int' and 'integer' types ? https://github.com/qgis/QGIS/blob/release-2_18/src/server/qgswfsprojectparser.cpp#L473

Comment (rldhont) [reply lizmap/www/js/attributeTable.js]: Yes, the twice are possible and all other integer types.

File: lizmap/www/js/attributeTable.js [hunk 2]
```diff
@@ -1297,48 +1297,55 @@ var lizAttributeTable = function() {
...
+                            case 'integer':
+                            case 'int':
+                            case 'unsignedInt':
+                            case 'long':
+                            case 'unsignedLong':
+                                colConf['mRender'] = function( data, type, full, meta ){
+                                    return parseInt(data);
+                                }
+                                break;
+                            // case 'double':
```
Comment (rldhont) [inline lizmap/www/js/attributeTable.js]: Can you conserve double to avoid regression ?
Thanks

**Comment by josemvm** [comment]
@rldhont and @nboisteault thanks a lot for fix it!
however, the hyperlink was lost to media columns (for some layers, and i don´t know why...)
instead of the alias field, the path appears: media/xxxxxxxxx_imagens/a_as_4477.jpg

**Comment by josemvm** [comment]
@rldhont and @nboisteault i have found the problem after this commit: hyperlink was lost to media columns, for layers whose name contains blank spaces!

i can open a new issue if you prefer

**Comment by rldhont** [comment]
@josemvm thanks to open a new issue

**Pull request merged**